include/bluetooth/bluetooth.h

```diff
@@ -404,6 +539,150 @@ int bt_le_adv_update_data(const struct bt_data *ad, size_t ad_len,
+ *
+ *  @param[in] param Advertising parameters.
+ *  @param[in] cb    Callback struct to notify about advertiser activity. Can be
+ *                   NULL. Must point to valid memory during the lifetime of the
+ *                   advertising set.
+ *  @param[out] adv  Valid advertising set object on success or NULL otherwise.
+ *
+ *  @return Zero on success or (negative) error code otherwise.
+ */
+int bt_le_ext_adv_create(const struct bt_le_adv_param *param,
+			 const struct bt_le_ext_adv_cb *cb,
+			 struct bt_le_ext_adv **adv);
+
+struct bt_le_ext_adv_start_param {
+	/** Advertiser timeout (N * 10 ms).
+	 *
+	 *  Application will be notified by the advertiser sent callback.
+	 *  Set to zero for no timeout.
+	 *
+	 *  If privacy :option:`CONFIG_BT_PRIVACY` is enabled then the timeout
+	 *  must be less than :option:`CONFIG_BT_RPA_TIMEOUT`.
+	 */
+	u16_t timeout;
+	/** Number of advertising events.
+	 *
+	 *  Application will be notified by the advertiser sent callback.
+	 *  Set to zero for no limit.
+	 */
+	u8_t  num_events;
+};
+
+/** Start advertising with the given advertising set
+ *
+ *  If the advertiser is limited by either the timeout or number of advertising
+ *  events the application will be notified by the advertiser sent callback once
+ *  the limit is reached.
+ *  If the advertiser is limited by both the timeout and the number of
+ *  advertising events then the limit that is reached first will stop the
+ *  advertiser.
+ *
+ *  @param adv    Advertising set object.
+ *  @param param  Advertise start parameters.
+ */
+int bt_le_ext_adv_start(struct bt_le_ext_adv *adv,
+			struct bt_le_ext_adv_start_param *param);
+
+/** Stop advertising with the given advertising set
+ *
+ *  Stop advertising with a specific advertising set. When using this function
+ *  the advertising sent callback will not be called.
+ *
+ *  @param adv Advertising set object.
+ *
+ *  @return Zero on success or (negative) error code otherwise.
+ */
+int bt_le_ext_adv_stop(struct bt_le_ext_adv *adv);
+
+/** Set advertising set advertising or scan response data.
+ *
+ *  Set advertisement data or scan response data. If the advertising set is
+ *  currently advertising then the advertising data will be updated in
+ *  subsequent advertising events.
+ *
+ *  If the advertising set has been configured to send advertising data on the
+ *  primary advertising channels then the maximum data length is
+ *  @ref BT_GAP_ADV_MAX_ADV_DATA_LEN bytes.
+ *  If the advertising set has been configured for extended advertising,
+ *  then the maximum data length is defined by the controller with the maximum
+ *  possible of @ref BT_GAP_ADV_MAX_EXT_ADV_DATA_LEN bytes.
+ *
+ *  @note Not all scanners support extended data length advertising data.
+ *
+ *  @note When updating the advertising data while advertising the advertising
+ *        data and scan response data must be less than 251 bytes. Otherwise the
```

Comment:
This 251 is wrong, remove it for now.

Reply:
Replace with "data length must be smaller or equal to what can be fit in a single advertising packet`.